doc/user_guide/big_data.rst

```diff
@@ -0,0 +1,228 @@
+might look like:
+
+.. code-block:: python
+
+    >>> s = hs.load("shish26.02-6.blo", lazy=True)
+    >>> s
+    <LazySignal2D, title: , dimensions: (400, 333|512, 512)>
+    >>> s.data
+    dask.array<array-e..., shape=(333, 400, 512, 512), dtype=uint8, chunksize=(20, 12, 512, 512)>
+    >>> print(s.data.dtype, s.data.nbytes / 1e9)
+    uint8 34.9175808
+    >>> s.change_dtype("float") # To be able to perform decomposition, etc.
+    >>> print(s.data.dtype, s.data.nbytes / 1e9)
+    float64 279.3406464
+
+Only to load the data in the original unsigned integer format it would require
+around 35GB of memory. To store it in a floating-point format one would need
+almost 280GB of memory. Nevertheless, with the lazy processing both of these
+steps are near instantaneous and require very little resources.
+
+Lazy stacking
+^^^^^^^^^^^^^
+
+Occasionally the full dataset consists of many smaller files. To combine them
+into a one large ``LazySignal``, we can :ref:`stack<signal.stack_split>` them
+lazily (both when loading of afterwards):
+
+.. code-block:: python
+
+    >>> siglist = hs.load("*.hdf5")
+    >>> s = hs.stack(siglist, lazy=True)
+    >>> # Or do that when loading
+    >>> s = hs.load("*.hdf5", lazy=True, stack=True)
+
+Casting signals as lazy
+^^^^^^^^^^^^^^^^^^^^^^^
+
+To convert a regular HyperSpy signal to a lazy one (so that any future
+operations are only performed lazily) we added
+:py:meth:`~.signal.BaseSignal.as_lazy` method:
+
+.. code-block:: python
+
+    >>> s = hs.signals.Signal1D(np.arange(150.).reshape((3, 50)))
+    >>> s
+    <Signal1D, title: , dimensions: (3|50)>
+    >>> sl = s.as_lazy()
+    >>> sl
+    <LazySignal1D, title: , dimensions: (3|50)>
+
+
+
+Constraints
+-----------
+
+A couple of constraints to ``LazySignal`` when compared to normal HyperSpy
+signals.
+
+Immutable signals
+^^^^^^^^^^^^^^^^^
+
+One of the most often encountered constraint when using ``LazySignal`` is the
+inability to modify an existing signal data (immutability). It is a logical
+consequence of the DAG (tree structure), where a complete history of the
+processing has to be stored to traverse later. 
+
+In fact, ``LazySignal`` removes the need for such operation, since only
+additional tree branches are added, requiring very little resources. In
+practical terms:
+
+.. code-block:: python
+
+    >>> s = s + 1 # instead of s += 1
+    >>> # or, even better
+    >>> s1 = s + 1
+
+Machine learning (decomposition)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:ref:`Machine learning<ml>` often performs large matrix manipulations,
+requiring significantly more memory than just storing the data. Lazy HyperSpy
+signals attempted to provide similar alternatives. These algorithms read the
+data in an "online" manner (i.e. only loading each element once or twice),
+enabling performing the decompositions on very large datasets. All of the
+algorithms are only approximations to the original implementations, some
+(ONMF in particular) sparsely tested.
```

Comment:
This isn't true - ORPCA is taken directly from the original implementation with some additions and improvements for faster processing!